velodrome/transform/fetcher.go

```diff
@@ -26,103 +27,108 @@ import (
 )
 
 // fetchRecentIssues retrieves issues from DB, but only fetches issues modified since last call
-func fetchRecentIssues(db *gorm.DB, last time.Time, out chan sql.Issue) (time.Time, error) {
-	glog.Infof("Fetching issues updated after %s", last)
+func fetchRecentIssues(db *gorm.DB, last *time.Time, out chan sql.Issue) error {
+	glog.Infof("Fetching issues updated after %s", *last)
 
 	var issues []sql.Issue
 	query := db.Where("issue_updated_at >= ?", last).Order("issue_updated_at").Preload("Labels").Find(&issues)
 	if query.Error != nil {
-		return last, query.Error
+		return query.Error
 	}
 
 	count := len(issues)
 	for _, issue := range issues {
 		out <- issue
-		last = issue.IssueUpdatedAt
+		*last = issue.IssueUpdatedAt
 	}
 	glog.Infof("Found and pushed %d updated/new issues", count)
 
-	return last, nil
+	return nil
 }
 
-// fetchRecentEvents retrieves events from DB, but only fetches events created since last call
-func fetchRecentEvents(db *gorm.DB, last int, out chan sql.IssueEvent) (int, error) {
-	glog.Infof("Fetching issue-events with id bigger than %d", last)
+// fetchRecentEventsAndComments retrieves events from DB, but only fetches events created since last call
+func fetchRecentEventsAndComments(db *gorm.DB, lastEvent *int, lastComment *int, out chan interface{}) error {
```

Comment:
I've improved the test to make sure it merges properly (we end-up with sorted time). I'm not super fan of breaking this function into pieces.